pkg/test/framework/components/namespace/dump.go

```diff
@@ -0,0 +1,180 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package namespace
+
+import (
+	"fmt"
+	"github.com/gogo/protobuf/jsonpb"
+	"io/ioutil"
+	"istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/scopes"
+	"k8s.io/api/core/v1"
+	"os"
+	"path"
+)
+
+var defaultDumpers = []podDumper{
+	DumpPodState,
+	DumpPodEvents,
+	DumpPodLogs,
+	DumpPodProxies,
+}
+
+// podDumper will dump information from all the pods into the given workDir.
+// If no pods are provided, accessor will be used to fetch all the pods in a namespace.
+type podDumper func(workDir string, namespace string, accessor *kube.Accessor, pods ...v1.Pod)
+
+// DumpPods runs each dumper with all the pods in the given namespace.
+// If no dumpers are provided, their resource state, events, container logs and Envoy information will be dumped.
+func DumpPods(workDir, namespace string, accessor *kube.Accessor, dumpers ...podDumper) {
```

Comment:
In retrospect, the `dumpers ...podDumper` might not be necessary:
1. The thought was to avoid redundant GetPods calls
2. Is there a case where we don't want to dump everything possible? I tried to keep from changing too much existing behavior as is but I don't see the harm in dumping it all, every time.